### DISCUSSION TITLE
Dependencies Discussion

### DISCUSSION THREAD

**Discussion by 2320sharon** [SatelliteShorelines/CoastSeg#26] [Ideas]
# Dependencies Discussion
In order to place limits on the size of the bounding box drawn by the user, I need to calculate the area of the geojson bounding box.  I found a python package called "area" that uses the same method as geojson.io to calculate the area in sq. meters of a geojson polygon. The only problem is that it is not being actively maintained. 

Can I add this as a dependency or should I find some way to utilize the geopandas area method which requires some reprojection and CRS assistance?

## Links
https://pypi.org/project/area/
https://github.com/scisco/area

## Current Geopandas Geodataframe Info

<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

UserWarning: Geometry is in a geographic CRS. Results from 'area' are likely incorrect. Use 'GeoSeries.to_crs()' to re-project geometries to a projected CRS before this operation.

**Comment by ddbuscombe** [comment]
I'd say use `area`. We dont need a lot of precision here, so it seems like unnecessary work to use geopandas and CRS transformations. 

Looks like `area` hasnt been updated, but looking at the [implementation](https://github.com/scisco/area/blob/master/area/__init__.py), I doubt it would ever need updating

**Reply by 2320sharon** [reply]
Great, I'm glad we are on the same page. I will add it as a dependency and use it to limit the size of the bounding boxes in the code.

**Reply by kvos** [reply]
hi @2320sharon , you can also use `shapely` to calculate the area of a polygon (after reading the geojson with geopandas for example), this package is included in the current coastsat env fiile.

**Comment by 2320sharon** [comment]
# Removing Bloated Dependencies
At the moment CoastSeg has more dependencies than it actually uses. Can we trim a few off, so when its time for users to download and install the environment, it is not excessively large?

## Screenshot of the dependency plan.
![Coast Seg Diagram-Dependencies drawio](https://user-images.githubusercontent.com/61564689/151583840-7565be06-eee2-460c-832f-82ebdfc32ff7.png)
- I would like to remove "cartopy" and "simplekml" because they are currently not being utilized 
- I would also like to potentially remove dependencies shown in red because they are included as dependencies in other packages such as leafmap already having pandas
- Lastly, I think we can remove "localtileserver" because it is not currently being used

Feel free to leave any feedback and ideas. Trimming the dependencies doesn't need to be finalized until we are prepping for a final release.
